runtime/src/main/java/org/corfudb/runtime/collections/streaming/StreamingTask.java

```diff
@@ -76,20 +77,40 @@ public StreamingTask(CorfuRuntime runtime, ExecutorService workerPool, String na
         this.listenerId = String.format("listener_%s_%s_%s", listener, namespace, streamTag);
         this.listener = listener;
         TableRegistry registry = runtime.getTableRegistry();
-        final UUID streamId = TableRegistry.getStreamIdForStreamTag(namespace, streamTag);
+        final UUID streamId;
+        // Federated tables do not have a stream tag, only an option isFederated set to true. Internally,
+        // their stream tag is constructed when the table is opened. This stream tag is constructed differently than
+        // other tables. So for subscribers on this tag, do not construct the streamId using the generic algorithm.
+        if (streamTag.equals(ObjectsView.getLogReplicatorStreamId().toString())) {
+            streamId = ObjectsView.getLogReplicatorStreamId();
+        } else {
+            streamId = TableRegistry.getStreamIdForStreamTag(namespace, streamTag);
+        }
```

Comment:
This seems a bit overkill to me that we modify the StreamingTask to track lr txn stream for a test. Do we have alternative way to test it?